app/controllers/posts_controller.rb

```diff
@@ -12,9 +12,9 @@ def index
   def show; end
 
   def create
-    @post = current_user.posts.build(post_params)
+    @new_post = current_user.posts.build(post_params)
```

Comment:
why do you need to change it?

Reply:
Because I have some conflicts in modal window:
1) If I visit show or edit page (for post) and open modal window for creating new post at the same time, form in modal window will be filled with data of current post.
2) If i visit edit page (for post) and open modal window for creating new post and submit form on it, it will not create new post, it will update current post.